core/classes/Adapter.js

```diff
@@ -153,6 +173,11 @@ module.exports = class Adapter {
     }
     const fixedData = this._sanitize(identity, data)
 
+    const validation = this._validate(_.omit(this.schemas[identity], "required"), _.omitBy(fixedData, _.isNil))
```

Comment:
It's better to validate pseudo object created from patch and old data. If you validate only patch, some single-field changes will fail because they might miss some required properties.